src/server/minecraft-server.js

```diff
@@ -0,0 +1,1503 @@
+
+        if (properties.installed) {
+            await this.log('Getting MinecraftServer banned IPs...');
+
+            try {
+                properties.bannedIps = JSON.parse(await fs.readFile(path.join(minecraftDirectory, 'banned-ips.json'), 'utf8'));
+                return Promise.resolve();
+            } catch (err) {
+                properties.bannedIps = [];
+                await this.log('Failed to read banned-ips.json:');
+                await this.log(err);
+                return Promise.reject();
+            }
+        } else {
+            return Promise.reject('Minecraft not installed.');
+        }
+    }
+
+    async getBannedPlayers () {
+        let properties = this.properties;
+        let minecraftDirectory = properties.settings.minecraftDirectory;
+
+        if (properties.installed) {
+            await this.log('Getting MinecraftServer banned players...');
+
+            try {
+                properties.bannedPlayers = JSON.parse(await fs.readFile(path.join(minecraftDirectory, 'banned-players.json'), 'utf8'));
+                return Promise.resolve();
+            } catch (err) {
+                properties.bannedPlayers = [];
+                await this.log('Failed to read banned-players.json:');
+                await this.log(err);
+                return Promise.reject(err);
+            }
+        } else {
+            return Promise.reject('Minecraft not installed.');
+        }
+    }
+
+    getCommand (line, required, optional) {
+        let start = 0;
+        let end = 0;
+        let arg;
+        let args = [];
+        let startChar, endChar;
+
+        if (required) {
+            startChar = '<';
+            endChar = '>';
+        }
+        if (optional) {
+            startChar = '[';
+            endChar = ']';
+        }
+
+        while (start !== -1) {
+            start = line.indexOf(startChar, end);
+            end = line.indexOf(endChar, start);
+
+            if (start !== -1) {
+                let option = line.substr(start + 1, end - start - 1);
+                let options = option.concat(line.substr(end + 1, line.indexOf(' ', end + 1) - end - 1));
+                if (options.indexOf('|') !== -1) {
+                    let args2 = options.split('|');
+                    while ( (arg = args2.shift()) !== undefined ) {
+                        args.push(arg);
+                    }
+                } else {
+                    args.push(options);
+                }
+                start = line.indexOf(startChar, end);
+            }
+        }
+
+        return args;
+    }
+
+    async getEula () {
+        await this.log('Getting MinecraftServer EULA acceptance state...');
+
+        let eula;
+        let eulaUrlLine;
+        let line;
+        let lineNumber;
+        let properties = this.properties;
+        let minecraftDirectory = properties.settings.minecraftDirectory;
+
+        await this.log('Reading MinecraftServer eula.txt...');
+        try {
```

Comment:
`try/catch` blocks should always start at the very top of the function. That would get the `log` functions inside the as well
I've probably missed a ton of these